rust/arrow/src/array/equal_json.rs

```diff
@@ -49,6 +49,20 @@ impl<T: ArrowPrimitiveType> JsonEqual for PrimitiveArray<T> {
     }
 }
 
+/// Implement array equals for numeric type
+impl JsonEqual for BooleanArray {
+    fn equals_json(&self, json: &[&Value]) -> bool {
+        if self.len() != json.len() {
+            return false;
+        }
+
+        (0..self.len()).all(|i| match json[i] {
```

Comment:
Could be `self.len() == json.len() && (0..self.len()....`

Reply:
Just out of curiosity, do you recommend `self.len() == json.len() && ...` or `self.len() == json.len() & ...`? I read somewhere that Rust short-circuits bit operators.

Reply:
I just thought `self.len() == json.len() &&` would read better than `if (self.len() == json.len()) return false ` in this case.
For short-circuiting I'm not sure? I was assuming Rust short-circuits `&&` and not `&`, that would be the "unsurprising" choice to me but I might be wrong.

Reply:
you are right, && does, & does not: https://doc.rust-lang.org/reference/expressions/operator-expr.html#lazy-boolean-operators